src/acmApi/authHelper.ts

```diff
@@ -0,0 +1,31 @@
+import axios, {AxiosInstance, AxiosRequestConfig} from "axios";
+
+function getBaseApi() {
+    return axios.create({
+        baseURL: process.env.REACT_APP_ACMAPI_URL,
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+    });
+}
+
+function getAuthorizedApi(getAccessTokenSilently: () => Promise<string>): AxiosInstance {
+    const api = getBaseApi();
+    api.interceptors.request.use(async (config: AxiosRequestConfig) => {
+        const token = await getAccessTokenSilently();
+        config.headers.Authorization = `Bearer ${token}`
+        return config;
+    });
+    return api;
+}
+
+type Authorized<F> =
+    F extends (axios: AxiosInstance, ...r: infer R) => Promise<infer O>
+        ? (...r: R) => Promise<O>
```

Comment:
Just to clarify what's happening here... lmk if this is correct

We have a type called Authorized for a function.
The function needs to match the requirements above which states that the first variable must be an axiosInstance followed by any number of additional arguments... and must return a promise.
We are inferring the types of parameters and the result of the promise in `R` and `O` respectively to be used after the ternary operator

My only suggestion for this would be to use the words `Function`, `Parameters` & `Object` or something along those lines instead of `F`, `R` and `O` even though its typically convention to use single letters to denote generics

Reply:
Yep, that's exactly right. I agree, I'll rename them.